collectors/python.d.plugin/mysql/mysql.chart.py

```diff
@@ -663,6 +734,65 @@ def userstats_chart_template(name):
 DEFAULT_REPL_CHANNEL = ''
 
 
+# Write Set REPlication
+# https://galeracluster.com/library/documentation/galera-status-variables.html
+# https://www.percona.com/doc/percona-xtradb-cluster/LATEST/wsrep-status-index.html
+class WSRepDataConverter:
+    unknown_value = -1
+
+    def convert(self, key, value):
+        if key == 'wsrep_connected':
+            return self.convert_connected(value)
+        elif key == 'wsrep_ready':
+            return self.convert_ready(value)
+        elif key == 'wsrep_cluster_status':
+            return self.convert_cluster_status(value)
+        elif key == 'wsrep_local_state_comment':
+            return self.convert_local_state_comment(value)
+        return value
+
+    def convert_connected(self, value):
+        # https://www.percona.com/doc/percona-xtradb-cluster/LATEST/wsrep-status-index.html#wsrep_connected
+        if value == 'OFF':
+            return 0
+        if value == 'ON':
+            return 1
+        return self.unknown_value
+
+    def convert_ready(self, value):
+        # https://www.percona.com/doc/percona-xtradb-cluster/LATEST/wsrep-status-index.html#wsrep_ready
+        if value == 'OFF':
+            return 0
+        if value == 'ON':
+            return 1
+        return self.unknown_value
+
+    def convert_local_state_comment(self, value):
```

Comment:
In this function you are testing values in the range [0,3], but the official documentation uses value between [1,4], considering that I do not have a galera cluster, I would like to confirm that this is not creating problems in possible reports, can you confirm this?

Reply:
> https://www.percona.com/doc/percona-xtradb-cluster/LATEST/wsrep-status-index.html#wsrep_local_state_comment

I wouldnt say it is official documentaion. There is just a table with numbers. 

Probably we need to use

> wsrep_local_state
> Internal Galera Cluster FSM state number.

Without decoding text to int.


https://github.com/codership/galera/blob/f2164433df3830e02adf319d1b0c4a8524a7e0bc/galera/src/replicator_smm_stats.cpp#L11-L24

I was googling for `galera source code` and found this

https://github.com/codership/galera/blob/f2164433df3830e02adf319d1b0c4a8524a7e0bc/galera/src/replicator_smm_stats.cpp#L11-L24

```cpp
static const char* state_str[GALERA_STAGE_MAX] =
{
    "Initialized",
    "Joining",
    "Joining: preparing for State Transfer",
    "Joining: requested State Transfer",
    "Joining: receiving State Transfer",
    "Joined",
    "Synced",
    "Donor/Desynced",
    "Joining: State Transfer request failed",
    "Joining: State Transfer failed",
    "Destroyed"
};
```

Reply:
Thanks!

Reply:
Probably you are right

> Internal number and the corresponding human-readable comment of the node’s state. Possible values are:

> Internal number

So it is `[1-4]`

Reply:
I am glad I could help you! :)